File: websocket_test_server.py
#!/usr/bin/env python3
"""
简单的 WebSocket 测试服务器
用于测试 Quest 应用的 Wi-Fi 连接功能

使用方法：
1. 安装依赖：pip install websockets
2. 运行：python websocket_test_server.py
3. 确保 PC/手机和 Quest 在同一 Wi-Fi 网络
4. 修改代码中的 IP 地址为 PC/手机的 IP
"""

import asyncio
import websockets
import json
from datetime import datetime

# 服务器配置
HOST = "0.0.0.0"  # 监听所有网络接口
PORT = 8080

connected_clients = set()

async def handle_client(websocket, path):
    """处理客户端连接"""
    client_addr = websocket.remote_address
    print(f"[{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] 客户端已连接: {client_addr}")
    connected_clients.add(websocket)
    
    try:
        # 发送欢迎消息
        await websocket.send("服务器已连接，可以开始通信")
        print(f"已向 {client_addr} 发送欢迎消息")
        
        # 接收消息
        async for message in websocket:
            print(f"[{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] 收到消息: {message}")
            
            # 回显消息
            response = f"服务器收到: {message}"
            await websocket.send(response)
            print(f"已回复: {response}")
            
    except websockets.exceptions.ConnectionClosed:
        print(f"[{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] 客户端断开连接: {client_addr}")
    except Exception as e:
        print(f"错误: {e}")
    finally:
        connected_clients.discard(websocket)

async def main():
    """启动服务器"""
    print("=" * 60)
    print("WebSocket 测试服务器")
    print("=" * 60)
    print(f"监听地址: ws://{HOST}:{PORT}")
    print(f"本地访问: ws://localhost:{PORT}")
    print(f"局域网访问: ws://<你的IP>:{PORT}")
    print("=" * 60)
    print("等待客户端连接...")
    print("按 Ctrl+C 停止服务器")
    print("=" * 60)
    
    async with websockets.serve(handle_client, HOST, PORT):
        await asyncio.Future()  # 永久运行

if __name__ == "__main__":
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        print("\n服务器已停止")

File: websocket_test_server.js
/**
 * 简单的 WebSocket 测试服务器 (Node.js 版本)
 * 
 * 使用方法：
 * 1. 安装 Node.js
 * 2. 安装依赖：npm install ws
 * 3. 运行：node websocket_test_server.js
 */

const WebSocket = require('ws');

const PORT = 8080;
const server = new WebSocket.Server({ 
    host: '0.0.0.0',  // 监听所有网络接口
    port: PORT 
});

console.log('='.repeat(60));
console.log('WebSocket 测试服务器 (Node.js)');
console.log('='.repeat(60));
console.log(`监听地址: ws://0.0.0.0:${PORT}`);
console.log(`本地访问: ws://localhost:${PORT}`);
console.log(`局域网访问: ws://<你的IP>:${PORT}`);
console.log('='.repeat(60));
console.log('等待客户端连接...');
console.log('按 Ctrl+C 停止服务器');
console.log('='.repeat(60));

server.on('connection', (ws, req) => {
    const clientIP = req.socket.remoteAddress;
    const timestamp = new Date().toLocaleString('zh-CN');
    
    console.log(`[${timestamp}] 客户端已连接: ${clientIP}`);
    
    // 发送欢迎消息
    ws.send('服务器已连接，可以开始通信');
    console.log(`已向 ${clientIP} 发送欢迎消息`);
    
    // 接收消息
    ws.on('message', (message) => {
        const msg = message.toString();
        const timestamp = new Date().toLocaleString('zh-CN');
        console.log(`[${timestamp}] 收到消息: ${msg}`);
        
        // 回显消息
        const response = `服务器收到: ${msg}`;
        ws.send(response);
        console.log(`已回复: ${response}`);
    });
    
    // 处理断开连接
    ws.on('close', () => {
        const timestamp = new Date().toLocaleString('zh-CN');
        console.log(`[${timestamp}] 客户端断开连接: ${clientIP}`);
    });
    
    // 处理错误
    ws.on('error', (error) => {
        console.error(`错误: ${error.message}`);
    });
});

// 优雅关闭
process.on('SIGINT', () => {
    console.log('\n正在关闭服务器...');
    server.close(() => {
        console.log('服务器已停止');
        process.exit(0);
    });
});


